src/js/components/ServicePlanBlocks.js

```diff
@@ -0,0 +1,195 @@
+import classNames from 'classnames';
+import {Tooltip} from 'reactjs-components';
+import React from 'react';
+
+import IconUpgradeBlock from './icons/IconUpgradeBlock';
+import IconUpgradeBlockDecisionPoint from './icons/IconUpgradeBlockDecisionPoint';
+import ServicePlanBlock from '../structs/ServicePlanBlock';
+import ServicePlanStore from '../stores/ServicePlanStore';
+
+const METHODS_TO_BIND = [
+  'handleExternalClick',
+  'handleBlockMouseEnter',
+  'handleBlockMouseLeave'
+];
+
+class ServicePlanBlocks extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      detailsExpanded: false,
+      hoveredBlock: null,
+      selectedBlock: null
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentDidMount() {
+    window.addEventListener('click', this.handleExternalClick);
+  }
+
+  handleBlockClick(block, event) {
+    // Stop event from propagating to window, which would dismiss the tooltip
+    // that we're trying to open.
+    event.stopPropagation();
+
+    this.handleExternalClick();
+    this.setState({selectedBlock: block});
+    this.refs[block.getID()].triggerOpen();
+  }
+
+  handleExternalClick() {
+    if (!!this.state.selectedBlock) {
+      this.refs[this.state.selectedBlock.getID()].triggerClose();
+      this.setState({selectedBlock: null});
+    }
+  }
+
+  handleBlockRestart(blockID, phaseID) {
+    ServicePlanStore.restartBlock(this.props.service.id, blockID, phaseID);
+  }
+
+  handleBlockForceComplete(blockID, phaseID) {
+    ServicePlanStore.forceCompleteBlock(this.props.service.id, blockID,
+      phaseID);
+  }
+
+  handleBlockMouseEnter(block) {
+    this.setState({hoveredBlock: block});
+  }
+
+  handleBlockMouseLeave() {
+    this.setState({hoveredBlock: null});
+  }
+
+  getUpgradeBlocks(phaseBlocks, activePhase) {
+    let blocks = phaseBlocks.getItems().map((block, blockIndex) => {
+      let {selectedBlock} = this.state;
+      block = new ServicePlanBlock(block);
```

Comment:
It's not an instance of `Block`, though I agree it should be. Let me see what's up in the corresponding struct.